### PR TITLE
Ignore some MTLCompiler failures in impeller unit tests

### PR DESCRIPTION
**Pull request by zanderso**
This also re-enables the tests.

Related https://github.com/flutter/flutter/issues/117122

### CODE REVIEW

File: testing/run_tests.py [hunk 1]
```diff
@@ -92,10 +100,17 @@ def run_cmd(
 
     print_divider('!')
 
-    raise Exception(
-        'Command "%s" exited with code %d.' %
-        (command_string, process.returncode)
-    )
+    allowed_failure = False
```
Comment (gaaclarke) [inline testing/run_tests.py:103]:
```suggestion
    is_allowed_failure = False
```

File: testing/run_tests.py [hunk 2]
```diff
@@ -92,10 +100,17 @@ def run_cmd(
...
+    for allowed_string in allowed_failure_output:
+      if (stdout and allowed_string in stdout) or (stderr and
+                                                   allowed_string in stderr):
+        allowed_failure = True
+
+    if not allowed_failure:
```
Comment (gaaclarke) [inline testing/run_tests.py:109]: We should probably log allowed failures, right?

**Review by jason-simmons — APPROVED** [review]

**Comment by zanderso** [comment]
One of the presubmit checks caught an instance of this and the new code in `run_tests.py` appears to be doing the right thing: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8786351834451420033/+/u/test:_Host_Tests_for_host_profile/stdout?format=raw

File: testing/run_tests.py [hunk 3]
```diff
@@ -52,19 +52,27 @@ def is_asan(build_dir):
 
 
 def run_cmd(
-    cmd, forbidden_output=None, expect_failure=False, env=None, **kwargs
+    cmd,
+    forbidden_output=None,
+    expect_failure=False,
+    env=None,
+    allowed_failure_output=None,
```
Comment (gaaclarke) [inline testing/run_tests.py:59]:
```suggestion
    failure_allow_list=None,
```

File: testing/run_tests.py [hunk 4]
```diff
@@ -52,19 +52,27 @@ def is_asan(build_dir):
...
 
   command_string = ' '.join(cmd)
 
   print_divider('>')
   print('Running command "%s"' % command_string)
 
   start_time = time.time()
-  stdout_pipe = sys.stdout if not forbidden_output else subprocess.PIPE
-  stderr_pipe = sys.stderr if not forbidden_output else subprocess.PIPE
+  collect_output = forbidden_output or allowed_failure_output
```
Comment (gaaclarke) [inline testing/run_tests.py:73]:
```suggestion
  should_collect_output = forbidden_output or allowed_failure_output
```

**Review by gaaclarke — APPROVED** [review]
A couple nits about variable naming to make working with dynamic typing a bit easier and a note about logging allowed failures.  Otherwise, lgtm.

**Comment by gaaclarke** [comment]
Doh.

**Pull request merged**